storage/filesystem_storage.go

```diff
@@ -0,0 +1,140 @@
+package storage
+
+import (
+	"context"
+	"github.com/gazebo-web/gz-go/v7"
+	"github.com/pkg/errors"
+	"os"
+)
+
+// fsStorage contains the implementation of a storage manager for resources using the host filesystem.
+// It can be used with AWS EFS storage in EC2 instances.
+type fsStorage struct {
+	basePath string
+}
+
+func (s *fsStorage) UploadZip(ctx context.Context, resource Resource, file *os.File) error {
```

Comment:
I'm not, `Download` already generates a zip file in the `fsStorage` implementation, we can add it if you see the value on it.

Reply:
Having methods do different things depending on the implementation breaks the interface abstraction. If in my system I use the `Storage` interface, I expect to be able to change implementations without having to change the code, barring implementation configuration settings.

By this I mean that if I initially use the `S3v1` implementation, my code will have to explicitly upload a zip of the file before it is available for download. If I then switch to this implementation, my code will break because it will try to upload a zip file first, as is implied by the interface definition. Consider moving the zipping logic from the `Download` method to this method 

Note that I understand that having this implementation is not critical for us because we are unlikely to use it, but it is something to take into account when defining and implementing interfaces in general.

Reply:
That's a very valid comment, and I agree that all implementations should share the same underlying behavior. The original plan was to just have a `Download` method, and don't allow applications to upload zip versions of a resource. The reason why this change in the middle of the development phase is because `S3` doesn't allow you to download a set of files as a Zip file, and should be done manually by the consumer.

Considering that, I'll proceed to move the zip logic to the UploadZip fn.